### PR TITLE
net-vpc module: added subnet mini-factory

### PR DESCRIPTION
**Pull request by sruffilli**
This PR integrates a "mini-factory" to drive subnet creation from well-formed YaML files.

### CODE REVIEW

**Comment by sruffilli** [comment]
Still missing tests, don't merge!

File: modules/net-vpc/main.tf [hunk 1]
```diff
@@ -77,6 +78,39 @@ locals {
...
+  }
+
+  subnet_descriptions = merge(var.subnet_descriptions, local.subnet_data_descriptions)
+
+  subnet_data_private_access = {
+    for k, v in local._subnet_data : "${v.region}/${k}" => try(v.private_ip_google_access, true)
+  }
+
+  subnet_private_access = merge(var.subnet_private_access, local.subnet_data_private_access)
+
```
Comment (sruffilli) [inline modules/net-vpc/main.tf:113]: Missing IAM bindings from YaML

**Comment by sruffilli** [comment]
Ready for review @ludoo

File: modules/net-vpc/main.tf [hunk 2]
```diff
@@ -77,6 +67,58 @@ locals {
...
+    trimsuffix(basename(f), ".yaml") => merge(
+      yamldecode(file("${var.data_folder}/${f}")),
+      {
+        project_id = var.project_id
+        network    = local.network.name
+      }
+    )
+  }
+
+  subnet_data = var.data_folder == null ? {} : {
```
Comment (ludoo) [inline modules/net-vpc/main.tf:82]: you already check for `var.data_folder` in `_subnet_data`

Comment (ludoo) [reply modules/net-vpc/main.tf:82]: Do we really need a second step? The variables are all there in `_subnet_data` already. I would move the `try` in the resource, and do away with most of these locals.

**Review by ludoo — APPROVED** [review]

File: modules/net-vpc/main.tf (hunk 2 above)
Comment (sruffilli) [reply modules/net-vpc/main.tf:82]: Ack - cleaned check in `subnet_data`

I reckon we do need the second step, as yamldecode doesn't allow us to further manipulate datafields. Might be wrong.

Comment (sruffilli) [reply modules/net-vpc/main.tf]: Resolved in a0f1593

**Pull request merged**